tests/lib/rules/prefer-named-capture-group.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * @fileoverview Tests for prefer-named-capture-group rule.
+ * @author Pig Fang <https://github.com/g-plane>
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/prefer-named-capture-group"),
+    RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 2018 } });
+
+ruleTester.run("prefer-named-capture-group", rule, {
+    valid: [
+        "/normal_regex/",
+        "/(?:[0-9]{4})/",
+        "/(?<year>[0-9]{4})/",
+        "/\\u{1F680}/u",
+        "new RegExp()",
+        "new RegExp(foo)",
+        "new RegExp('')",
+        "new RegExp('(?<year>[0-9]{4})')",
+        "RegExp()",
+        "RegExp(foo)",
+        "RegExp('')",
+        "RegExp('(?<year>[0-9]{4})')",
+        "RegExp('(')", // invalid regexp should be ignored
+        "RegExp('\\\\u{1F680}', 'u')"
+    ],
+
+    invalid: [
```

Comment:
Sometimes, people use parentheses to indicate grouping in a regex without using it as a capture group, e.g.

```js
const regex = /(ab|cd|ef)+/;
```

Would this report an error? It could be replaced with `/(?:ab|cd|ef)+/`, although maybe some users would think that is unnecessary.

If it would report an error, maybe we should modify the message to avoid telling the user to use a named capture group when they can just use a non-capturing group.